File: backend_golang/presentation/handler/get_all_todos_handler.go
package handler

import (
	"backend_golang/application_service/usecase"
	"net/http"

	"github.com/google/uuid"
	"github.com/labstack/echo/v4"
)

type GetAllTodosHandler struct {
	au usecase.GetAllTodosUsecase
}

func NewGetAllTodosHandler(au usecase.GetAllTodosUsecase) HTTPHandler {
	return &GetAllTodosHandler{
		au: au,
	}
}

type GetAllTodosTodoDTO struct {
	ID          uuid.UUID `json:"id"`
	Title       string    `json:"title"`
	Description *string   `json:"description"`
	Completed   bool      `json:"completed"`
}

type GetAllTodosResponse struct {
	Todos []GetAllTodosTodoDTO `json:"todos"`
}

func (h *GetAllTodosHandler) Handle(c echo.Context) error {
	todos, err := h.au.Handle(c.Request().Context())
	if err != nil {
		return echo.NewHTTPError(http.StatusInternalServerError, err.Error())
	}
	response := GetAllTodosResponse{
		Todos: make([]GetAllTodosTodoDTO, len(todos)),
	}
	for i, todo := range todos {
		t := todo.Serialize()
		response.Todos[i] = GetAllTodosTodoDTO{
			ID:          t.ID,
			Title:       t.Title,
			Description: t.Description,
			Completed:   t.Completed,
		}
	}
	return c.JSON(http.StatusOK, response)
}

File: backend_python/src/todo_api/presentation/handler/get_all_todos_handler.py
from pydantic import BaseModel

from todo_api.application_service.usecase.get_all_todos_usecase import (
    GetAllTodosUsecase,
    GetAllTodosUsecaseInput,
)


class TodoDTO(BaseModel):
    id: str
    title: str
    description: str | None = None
    completed: bool


class GetAllTodosResponse(BaseModel):
    todos: list[TodoDTO]


class GetAllTodosHandler:
    def __init__(self, get_all_todos_usecase: GetAllTodosUsecase) -> None:
        self.get_all_todos_usecase = get_all_todos_usecase

    def handle(self) -> GetAllTodosResponse:
        result = self.get_all_todos_usecase.execute(GetAllTodosUsecaseInput())
        todos = [
            TodoDTO(
                id=str(todo.id),
                title=todo.title,
                description=todo.description,
                completed=todo.completed,
            )
            for todo in result.todos
        ]
        return GetAllTodosResponse(todos=todos)
